R/util.R

```diff
@@ -1474,3 +1474,118 @@ one_hot <- function(df, key) {
   # Spread the column into multiple columns with name <original column name>_<original value> and value of 1 or 0.
   df %>% tidyr::spread(!!rlang::enquo(key), !!rlang::sym(tmp_value_col), fill = 0, sep = "_") %>% select(-!!rlang::sym(tmp_id_col))
 }
+
+
+#'Wrapper function for dplyr::bind_rows to support named data frames when it's called inside dplyr chain.
+#'@export
+bind_rows <- function(..., .id = NULL, first_id = '', ignore_column_data_type = TRUE) {
+  # If the dplyr::bind_rows is called within a dplyr chain like df1 %>% dplyr::bind_rows(list(df_2 = df2, df_3 = df3), .id="id"),
+  # since df1 does not have a name, the "id" column of the resulting data frame does not have the data frame name for rows from df1.
+  # To workaround this issue, set a name to the first data frame with the value specified by fistLabel argument as a pre-process
+  # then pass the updated list to dplyr::bind_rows.
+  x_updated <- list()
+  x <- dplyr:::flatten_bindable(rlang::dots_values(...))
+  if(ignore_column_data_type | stringr::str_length(first_id) >0) {
+    count <- 0;
+    if(!is.null(names(x))) {
```

Comment:
Explain about expected behavior for different ways of passing argument.

Reply:
added comment.